File: src/test/java/net/pterodactylus/sone/web/page/FreenetRequestTest.java
package net.pterodactylus.sone.web.page;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.mockito.Mockito.mock;

import java.net.URI;
import java.net.URISyntaxException;

import net.pterodactylus.util.web.Method;

import freenet.clients.http.ToadletContext;
import freenet.support.api.HTTPRequest;

import org.junit.Test;

/**
 * Unit test for {@link FreenetRequest}.
 */
public class FreenetRequestTest {

	private final URI uri = new URI(".");
	private final Method method = Method.GET;
	private final HTTPRequest httpRequest = mock(HTTPRequest.class);
	private final ToadletContext toadletContext = mock(ToadletContext.class);
	private final FreenetRequest request = new FreenetRequest(uri, method, httpRequest, toadletContext);

	@SuppressWarnings("unused")
	public FreenetRequestTest() throws URISyntaxException {
	}

	@Test
	public void uriIsRetainedCorrectly() {
		assertThat(request.getUri(), is(uri));
	}

	@Test
	public void methodIsRetainedCorrectly() {
		assertThat(request.getMethod(), is(method));
	}

	@Test
	public void httpRequestIsRetainedCorrectly() {
		assertThat(request.getHttpRequest(), is(httpRequest));
	}

	@Test
	public void toadletContextIsRetainedCorrectly() {
		assertThat(request.getToadletContext(), is(toadletContext));
	}

}

File: src/test/kotlin/net/pterodactylus/sone/web/page/FreenetRequestTest.kt
package net.pterodactylus.sone.web.page

import freenet.clients.http.*
import freenet.support.api.*
import net.pterodactylus.sone.test.*
import net.pterodactylus.util.web.*
import org.hamcrest.MatcherAssert.*
import org.hamcrest.Matchers.*
import org.junit.*
import org.mockito.Mockito.*
import java.net.*

class FreenetRequestTest {

	private val uri = URI(".")
	private val method = Method.GET
	private val httpRequest = mock(HTTPRequest::class.java)
	private val toadletContext = mock(ToadletContext::class.java)
	private val sessionManager = mock<SessionManager>()
	private val request = FreenetRequest(uri, method, httpRequest, toadletContext)

	@Test
	fun `uri is retained correctly`() {
		assertThat(request.uri, equalTo(uri))
	}

	@Test
	fun `method is retained correctly`() {
		assertThat(request.method, equalTo(method))
	}

	@Test
	fun `http request is retained correctly`() {
		assertThat(request.httpRequest, equalTo(httpRequest))
	}

	@Test
	fun `toadlet context is retained correctly`() {
		assertThat(request.toadletContext, equalTo(toadletContext))
	}

}
